roles/zabbix_agent/defaults/main.yml

```diff
@@ -229,8 +229,8 @@ zabbix_version_long: 5.2.4
 zabbix_win_package: zabbix_agent-{{ zabbix_version_long }}-windows-amd64-openssl.zip
 zabbix2_win_package: zabbix_agent2-{{ zabbix_version_long }}-windows-amd64-openssl-static.zip
 zabbix_win_download_url: https://cdn.zabbix.com/zabbix/binaries/stable
-zabbix_win_download_link: "{{ zabbix_win_download_url }}/{{ zabbix_version_long | regex_findall('[0-9]\\.[0-9]') }}/{{ zabbix_version_long }}/{{ zabbix_win_package }}"
-zabbix2_win_download_link: "{{ zabbix_win_download_url }}/{{ zabbix_version_long | regex_findall('[0-9]\\.[0-9]') }}/{{ zabbix_version_long }}/{{ zabbix2_win_package }}"
+zabbix_win_download_link: "{{ zabbix_win_download_url }}/{{ zabbix_version_long | regex_findall('[0-9]\\.[0-9]') | join('') }}/{{ zabbix_version_long }}/{{ zabbix_win_package }}"
```

Comment:
I think a cleaner way to accomplish this would be to use the regex_search filter and a better captured regex which will capture versions greater than 9. See below for an example


```yaml
zabbix_win_download_link: "{{ zabbix_win_download_url }}/{{ zabbix_version_long | regex_search('^\\d+\\.\\d+') }}/{{ zabbix_version_long }}/{{ zabbix_win_package }}"
zabbix2_win_download_link: "{{ zabbix_win_download_url }}/{{ zabbix_version_long | regex_search('^\\d+\\.\\d+') }}/{{ zabbix_version_long }}/{{ zabbix2_win_package }}"
```

Reply:
Please fix this bug to the main branch. Regex_search filter works fine.